app/src/main/java/org/mifos/selfserviceapp/ui/adapters/ClientChargeAdapter.java

```diff
@@ -46,21 +49,52 @@ public Charge getItem(int position) {
         return vh;
     }
 
+    // Binds the values for each of the stored variables to the view
+    // Also changes the color of the circle depending on whether the charge is active or not
     @Override
     public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
         if (holder instanceof RecyclerView.ViewHolder) {
 
             Charge charge = getItem(position);
-            ((ClientChargeAdapter.ViewHolder) holder).tvAmount.setText(
-                    String.valueOf(charge.getAmount() + charge.getCurrency().getCode()));
+            Currency c  = Currency.getInstance("" + (charge.getCurrency()).getCode());
+            ((ClientChargeAdapter.ViewHolder) holder).tvAmountDue.setText(
+                    String.valueOf(charge.getAmount() + " " + c.getSymbol()));
+            ((ClientChargeAdapter.ViewHolder) holder).tvAmountPaid.setText(
+                    String.valueOf(charge.getAmountPaid() + " " + c.getSymbol()));
+            ((ClientChargeAdapter.ViewHolder) holder).tvAmountWaived.setText(
+                    String.valueOf(charge.getAmountWaived() + " " + c.getSymbol()));
+            ((ClientChargeAdapter.ViewHolder) holder).tvAmountOutstanding.setText(
+                    String.valueOf(charge.getAmountOutstanding() + " " + c.getSymbol()));
             ((ViewHolder) holder).tvClientName.setText(charge.getName());
-            ((ViewHolder) holder).tvDueDate.setText(charge.getDueDate().get(2).toString() +
-                    Constants.BACK_SLASH + charge.getDueDate().get(1).toString() +
-                    Constants.BACK_SLASH + charge.getDueDate().get(0).toString());
+
+            if (charge.isIsActive()) {
+                ((ViewHolder) holder).circle_status
+                        .setImageDrawable(setCircularBackground(R.color.deposit_green));
+            } else if (charge.isIsPaid()) {
+                ((ViewHolder) holder).circle_status
+                        .setImageDrawable(setCircularBackground(R.color.light_yellow));
+            } else {
+                ((ViewHolder) holder).circle_status
+                        .setImageDrawable(setCircularBackground(R.color.light_blue));
+            }
+
+            String day = charge.getDueDate().get(2).toString();
+            String month = new DateFormatSymbols().getMonths()
+                            [Integer.parseInt(charge.getDueDate().get(1).toString()) - 1];
+            String year = charge.getDueDate().get(0).toString();
+
+            ((ViewHolder) holder).tvDueDate.setText(day + " " + month + " " + year);
         }
 
     }
 
+    private LayerDrawable setCircularBackground(int colorId) {
```

Comment:
I found this method is using multiple places, cam you please put this also in Utils.java and call here.